### PR TITLE
Fixed #23860 -- Documented import order convention.

### PR DESCRIPTION
**Pull request by timgraham**
To give a taste of the automated results: https://github.com/django/django/pull/4009

### CODE REVIEW

File: docs/internals/contributing/writing-code/coding-style.txt [hunk 1]
```diff
@@ -56,6 +48,81 @@ Python style
...
+      )
+
+      # local Django
+      from .models import LogEntry
+
+      # try/except
+      try:
+          import pytz
+      except ImportError:
+        pytz = None
```
Comment (wrwrwr) [inline docs/internals/contributing/writing-code/coding-style.txt]: I'd like two more spaces here :-)

**Comment by mjtamlyn** [comment]
Aside from the lack of trailing commas (https://github.com/timothycrosley/isort/issues/221) this looks like a good idea to me.

**Comment by wrwrwr** [comment]
Looks good. Some more [isort options](https://github.com/timgraham/django/blob/isort/setup.cfg#L10) from my old config to (potentially) consider:
```
skip = build,.git,./django/utils/dictconfig.py,./django/utils/unittest.py,./django/utils/lru_cache.py,./django/utils/six.py,./django/conf/app_template/*,./django/dispatch/weakref_backports.py
known_third_party = bcrypt
order_by_type = true
combine_as_imports = true
```
(``indent`` [defaults](https://github.com/timothycrosley/isort/blob/develop/isort/settings.py#L72) to 4 spaces, so you could skip it; ``order_by_type`` defaults to true since last month).

**Comment by timgraham** [comment]
re: `skip`, it didn't seem like any changes were made in those files anyway (and dictconfig/unittest are now removed on master). Doesn't hurt to add build/.git for any speed benefits I guess.

I couldn't see any benefit to `known_third_party` -- aren't all imports that aren't Django or stdlib third party?

Didn't know about `combine_as_imports`, I'll add it.

I'll remove `indent`.

**Comment by MarkusH** [comment]
From what I understand `known_third_party` are the packages that are sorted on the same level as `django`.

**Comment by collinanderson** [comment]
I'm using known_third_party because I have 3rd party libraries symlinked in my projects directory, and I don't want isort to those with my projects imports.

**Comment by timgraham** [comment]
This is ready for a final review (to ensure the docs are clear). I already merged the cleanup themselves.

File: docs/internals/contributing/writing-code/coding-style.txt [hunk 2]
```diff
@@ -48,6 +40,85 @@ Python style
...
+      try:
+          import pytz
+      except ImportError:
+          pytz = None
+
+      CONSTANT = 'foo'
+
+
+      class Example(object):
+        ...
```
Comment (MarkusH) [inline docs/internals/contributing/writing-code/coding-style.txt]: Indent by 4 and add leading `#`:
```python
class Example(object):
    # ...
```

**Pull request merged**